### PR TITLE
chore: add example for balance AccountStore

### PR DESCRIPTION
**Pull request by ECJ222**
Added an example for balance AccountStore #6532.

13g44iYtkxLZ2tkp6iTJiqs7yEiEr4bVsPZTueWDjd6Bor1P

### CODE REVIEW

File: frame/balances/src/lib.rs [hunk 1]
```diff
@@ -513,7 +513,11 @@ pub mod pallet {
 	pub type TotalIssuance<T: Config<I>, I: 'static = ()> = StorageValue<_, T::Balance, ValueQuery>;
 
 	/// The balance of an account.
-	///
+	/// ```
```
Comment (bkchr) [inline frame/balances/src/lib.rs:516]:
```suggestion
	/// # Example
	///
	/// ```
```

File: frame/balances/src/lib.rs [hunk 2]
```diff
@@ -513,7 +513,11 @@ pub mod pallet {
...
+	///  impl Balance for Runtime {
+	///    type AccountStore = StorageMapShim<Self::Account<Runtime>, frame_system::Provider<Runtime>, AccountId, Self::AccountData<Balance>>
```
Comment (bkchr) [inline frame/balances/src/lib.rs:518]: Some docs on what this does would be nice.

File: frame/balances/src/lib.rs [hunk 3]
```diff
@@ -513,7 +513,11 @@ pub mod pallet {
 	pub type TotalIssuance<T: Config<I>, I: 'static = ()> = StorageValue<_, T::Balance, ValueQuery>;
 
 	/// The balance of an account.
-	///
+	/// ```
+	///  impl Balance for Runtime {
```
Comment (kianenigma) [inline frame/balances/src/lib.rs:517]:
```suggestion
	///  impl pallet_balances::Config for Runtime {
```

**Review by kianenigma — CHANGES_REQUESTED** [review]
Needs addressing the review comments, thanks!

**Comment by bkchr** [comment]
Ping @ECJ222 ?

**Comment by ECJ222** [comment]
Hi @bkchr,

Hope you are doing good, I have been a little busy I will jump back on this issue.

**Comment by bkchr** [comment]
Can you please resolve the requested changes?

**Comment by ECJ222** [comment]
Hey @bkchr, 

I have resolved your requested changes.

File: frame/balances/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply frame/balances/src/lib.rs:497]: @ECJ222 this is still not done.

Comment (ECJ222) [reply frame/balances/src/lib.rs:497]: Please could you explain a little bit more on this, can I put the explanation above the updated code?

**Review by bkchr — APPROVED** [review]

File: frame/balances/src/lib.rs [hunk 4]
```diff
@@ -490,7 +490,13 @@ pub mod pallet {
 	pub type TotalIssuance<T: Config<I>, I: 'static = ()> = StorageValue<_, T::Balance, ValueQuery>;
 
 	/// The balance of an account.
+	/// # Example
 	///
+	/// ```
```
Comment (bkchr) [inline frame/balances/src/lib.rs:495]:
```suggestion
	/// ```nocompile
```

File: frame/balances/src/lib.rs (hunk 2 above)
Comment (kianenigma) [reply frame/balances/src/lib.rs:497]: sorry but doc still is not really sensible. You should explain the context, what options are available (store the data in this pallet or in system), which one is better, what tradeoff exists etc. 

I really don't see any value in adding this one example block.

Comment (ECJ222) [reply frame/balances/src/lib.rs:496]: I made an update to my initial pr @kianenigma

File: frame/balances/src/lib.rs [hunk 5]
```diff
@@ -488,8 +488,24 @@ pub mod pallet {
...
+	///  }
+	/// ```
+	/// You can also store the balance of an account in the System pallet.
+	/// # Example
+	/// ```nocompile
+	///  impl pallet_balances::Config for Runtime {
+	///   type AccountStore = system
+	///  }
+	/// ```
+	/// But this comes with tradeoffs, storing account balances in the system pallet stores
```
Comment (kianenigma) [inline frame/balances/src/lib.rs:506]:
```suggestion
	/// The Balances pallet example of storing the balance of an account.
	///
	/// # Example
	///
	/// ```nocompile
	///  impl pallet_balances::Config for Runtime {
	///    type AccountStore = StorageMapShim<Self::Account<Runtime>, frame_system::Provider<Runtime>, AccountId, Self::AccountData<Balance>>
	///  }
	/// ```
	/// 
	/// You can also store the balance of an account in the `System` pallet.
	///
	/// # Example
	///
	/// ```nocompile
	///  impl pallet_balances::Config for Runtime {
	///   type AccountStore = System
	///  }
	/// ```
	///
	/// But this comes with tradeoffs, storing account balances in the system pallet stores
```

File: frame/balances/src/lib.rs [hunk 6]
```diff
@@ -488,8 +488,24 @@ pub mod pallet {
...
+	/// frame_system data alongside the account data contrary to storing account balances in the
+	/// balances pallet, which uses a StorageMap to store balances data only.
```
Comment (kianenigma) [inline frame/balances/src/lib.rs:513]:
```suggestion
	/// `frame_system` data alongside the account data contrary to storing account balances in the
	/// `Balances` pallet, which uses a `StorageMap` to store balances data only.
```

**Review by kianenigma — APPROVED** [review]

**Comment by kianenigma** [comment]
bot merge

**Comment by kianenigma** [comment]
@ECJ222 please run `cargo +nightly fmt` and we're good to go.

**Comment by ECJ222** [comment]
> @ECJ222 please run `cargo +nightly fmt` and we're good to go.

I have formatted my code changes @kianenigma  👍

**Pull request merged**